src/cpr_sdk/yql_builder.py

```diff
@@ -94,15 +95,25 @@ def build_metadata_filter(self) -> Optional[str]:
         return None
 
     def build_concepts_filter(self) -> Optional[str]:
-        """Create the part of the query that limits to specific concepts"""
+        """
+        Create the part of the query that limits to specific concepts.
+
+        e.g:
+        - `concepts.name contains 'floods' and concepts.name contains 'environment'`
+        - `concepts.parent_concept_ids_flat matches 'Q123' and concepts.name contains 'environment'`
+        """
         if self.params.concept_filters:
             concepts_query = []
             for concept in self.params.concept_filters:
                 if concept.name == "parent_concept_ids_flat":
-                    concepts_query.append(f"{concept.name} matches '{concept.value}'")
+                    concepts_query.append(
+                        f"concepts.{concept.name} matches '{concept.value}'"
+                    )
                 else:
-                    concepts_query.append(f"{concept.name} contains '{concept.value}'")
-            return f"(concepts contains sameElement({', '.join(concepts_query)}))"
+                    concepts_query.append(
+                        f"concepts.{concept.name} contains '{concept.value}'"
+                    )
+            return f"({' and '.join(concepts_query)})"
```

Comment:
Just plugging this link we looked through to make sure this is what we want: 
- https://docs.vespa.ai/en/reference/query-language-reference.html#sameelement

> with normal AND the query persons.first_name AND persons.last_name will normally not give you what you want. It will match if a document has a persons element with a matching first_name AND any element with a matching last_name

So for the following test wouldn't we get any document passage with floods OR environment?
```
{"name": "name", "value": "floods"},
{"name": "name", "value": "environment"},
```